common/taskana-common/src/main/java/pro/taskana/common/internal/configuration/TaskanaConfigurationInitializer.java

```diff
@@ -161,6 +165,109 @@ Optional<T> initialize(
         TaskanaProperty taskanaProperty);
   }
 
+  static class MapPropertyParser implements PropertyParser<Map<?, ?>> {
+
+    @Override
+    public Optional<Map<?, ?>> initialize(
+        Map<String, String> properties,
+        String separator,
+        Field field,
+        TaskanaProperty taskanaProperty) {
+      if (!Map.class.isAssignableFrom(field.getType())) {
+        throw new SystemException(
+            String.format(
+                "Cannot initialize field '%s' because field type '%s' is not a Map",
+                field, field.getType()));
+      }
+
+      ParameterizedType genericType = (ParameterizedType) field.getGenericType();
+      Type[] actualTypeArguments = genericType.getActualTypeArguments();
+      Class<?> keyClass = (Class<?>) actualTypeArguments[0];
+      Type valueClass = actualTypeArguments[1];
+
+      // Parses property files into a Map using the following layout: <Property>.<Key> = <value>
+      String propertyKey = taskanaProperty.value();
+      Map<?, ?> mapFromProperties =
+          properties.keySet().stream()
+              .filter(it -> it.startsWith(propertyKey))
+              .map(
+                  it -> {
+                    // Keys of the map entry is everything after the propertyKey + "."
+                    String keyAsString = it.substring(propertyKey.length() + 1);
+                    Object key = getStringAsObject(keyAsString, keyClass);
+
+                    // Value of the map entry is the value from the property
+                    String propertyValue = properties.get(it);
+                    Object value = getStringAsObject(propertyValue, separator, valueClass);
+                    return Pair.of(key, value);
+                  })
+              .collect(Collectors.toUnmodifiableMap(Pair::getLeft, Pair::getRight));
```

Comment:
@mustaphazorgati  Since I learned via a failing test that Maps in TaskanaConfiguration should be unmodifiable, I thought it's a good idea to return an unmodifiable map here.